### PR TITLE
exclude kmeans feature columns

### PR DESCRIPTION
**Pull request by Yancey0623**
this PR implement:
1. add a new attribute `exclude_columns` on kmeans.
1. validate attribute in codegen.
 
In addition, also polish code of `fetch.go`.

### CODE REVIEW

File: cmd/sqlflowserver/main_test.go [hunk 1]
```diff
@@ -1392,20 +1392,21 @@ func CaseTrainPAIKMeans(t *testing.T) {
...
 	a.NoError(err)
 
-	trainSQL := fmt.Sprintf(`SELECT sepal_length,sepal_width,petal_length,petal_width FROM %s
+	trainSQL := fmt.Sprintf(`SELECT * FROM %s
 	TO TRAIN kmeans 
 	WITH
 		center_count=3,
-		idx_table_name=%s
+		idx_table_name=%s,
+		excluded_features=class
```
Comment (typhoonzero) [inline cmd/sqlflowserver/main_test.go]: excluded_features -> exclude_columns?

File: pkg/sql/codegen/pai/kmeans.go [hunk 2]
```diff
@@ -20,33 +20,73 @@ import (
...
+	"center_count": {attribute.Int, 3, `[default=3]
+The cluster count. range: [1, Infinity]
+`, attribute.IntLowerBoundChecker(1, true)},
+	"idx_table_name": {attribute.String, "", `
+The output table name which includes
+cluster_index column indicates the cluster result,
+distance column indicates the distance from the center and
+all the columns of input table.`, nil},
+	"excluded_columns": {attribute.String, "", `[default=""]
+excluded the special feature columns from the SELECT statment.`, nil},
```
Comment (weiguoz) [inline pkg/sql/codegen/pai/kmeans.go]: excluded -> Exclude
statment -> statement

File: pkg/sql/codegen/pai/kmeans.go [hunk 3]
```diff
@@ -20,33 +20,73 @@ import (
...
+func parseExcludedColsMap(attrs map[string]interface{}) (map[string]int, error) {
+	excludedColsMap := make(map[string]int)
+	excludedColsAttr := attrs["excluded_columns"].(string)
+	if excludedColsAttr != "" {
+		arr := strings.Split(excludedColsAttr, ",")
+		for _, e := range arr {
+			excludedColsMap[e] = 1
+		}
+	}
+	return excludedColsMap, nil
```
Comment (weiguoz) [inline pkg/sql/codegen/pai/kmeans.go]: It looks like the error(nil) in `return` is unnecessary.

File: pkg/sql/codegen/pai/kmeans.go (hunk 2 above)
Comment (Yancey0623) [reply pkg/sql/codegen/pai/kmeans.go]: `excluded` is adj.  I think it's ok.

File: pkg/sql/codegen/pai/kmeans.go (hunk 3 above)
Comment (Yancey0623) [reply pkg/sql/codegen/pai/kmeans.go]: That makes easy to do the unit test, but I think you're right, will add it if necessary.

**Review by weiguoz — APPROVED** [review]
LGTM

**Pull request merged**